src/OpalCompiler-Core/OCASTSemanticAnalyzer.class.st

```diff
@@ -116,11 +116,14 @@ OCASTSemanticAnalyzer >> outerScope: anObject [
 
 { #category : 'variables' }
 OCASTSemanticAnalyzer >> resolveVariableNode: aVariableNode [
+
 	| var |
-	var := (scope lookupVar: aVariableNode name)
-		ifNil: [ self undeclaredVariable: aVariableNode ].
+	var := (OCPseudoVariableScope new
+		        outerScope: scope;
```

Comment:
It should be really done at the level of scope construction. Then it can be easily overridden with special compiler configuration.

Reply:
Thanks we will check.

Reply:
Yes, I agree, but that is a feature outside of the scope of this PR.
We could do it afterwards if there is the need.

The good thing about this PR is that the resolution of pseudo-variables becomes part of the scope chain, before it was hardcoded in the global symbol resolution.

Reply:
One place where we need to take care is DoIts: in the DoIt, all the PseudoVariables are now wrapped in a DoItVariable which is then added to the scope to shadow the PseudoVariable.  This way when you inspect "thisContext" in the debugger, you get the context that the debugger looks at, not the context of the DoIt evaluation.

This means that we can not look up PseudoVariables as the very first thing, they have to be at least *after* the Scope that we add for the DoIt (this might already be the case, but it has to be checked).